include/swift/SILOptimizer/Analysis/ValueTracking.h

```diff
@@ -50,7 +50,8 @@ bool pointsToLocalObject(SILValue V);
 /// indirection (e.g. ref_element_addr, project_box, etc.).
 inline bool isUniquelyIdentified(SILValue V) {
   return pointsToLocalObject(V)
-         || isExclusiveArgument(getUnderlyingAddressRoot(V));
+         || (V->getType().isAddress()
```

Comment:
Was this a bug? Do we have a test case for this?

Reply:
No bug, it was just assumed that we were working with an address. I want to make that explicit everywhere and added asserts. A major part of the confusion is that some memory APIs accept random values and other expect an address for a specific access.